src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1953,150 +2119,35 @@ public static StackValue PerformReturnTypeCoerce(FunctionEndPoint functionEndPoi
-                    int resolutionFailures;
-
-                    Dictionary<FunctionEndPoint, int> lookups = funcGroup.GetExactMatchStatistics(
-                        context, reducedParams, stackFrame, runtimeCore,
-                        out resolutionFailures);
-
-
-                    if (resolutionFailures > 0)
-                        continue;
-
-                    return true; //Replicates against cluster
-                }
-            }
-
-            #endregion
-
-            #region Case 3: Match with type conversion, but no array promotion
-
-            {
-                Dictionary<FunctionEndPoint, int> candidatesWithDistances =
-                    funcGroup.GetConversionDistances(context, arguments, replicationControl.Instructions,
-                                                     runtimeCore.DSExecutable.classTable, runtimeCore);
-                Dictionary<FunctionEndPoint, int> candidatesWithCastDistances =
-                    funcGroup.GetCastDistances(context, arguments, replicationControl.Instructions, runtimeCore.DSExecutable.classTable,
-                                               runtimeCore);
-
-                List<FunctionEndPoint> candidateFunctions = GetCandidateFunctions(stackFrame, candidatesWithDistances);
-                FunctionEndPoint compliantTarget = GetCompliantTarget(context, arguments,
-                                                                      replicationControl.Instructions, stackFrame, runtimeCore,
-                                                                      candidatesWithCastDistances, candidateFunctions,
-                                                                      candidatesWithDistances);
-
-                if (compliantTarget != null)
-                {
-                    return false; //Type conversion but no replication
-                }
-            }
-
-            #endregion
-
-            #region Case 5: Match with type conversion, replication and array promotion
-
-            {
-                //Build the possible ways in which we might replicate
-                replicationTrials =
-                    Replicator.BuildReplicationCombinations(replicationControl.Instructions, arguments, runtimeCore);
-
-                //Add as a first attempt a no-replication, but allowing up-promoting
-                replicationTrials.Insert(0,
-                                         new List<ReplicationInstruction>()
-                    );
-            }
-
-            #endregion
-
-            return true; //It'll replicate if it suceeds
+            var helper = TraceSerialiserHelper.FromCallSiteData(callSiteData);
+            var serializables = helper.TraceData.SelectMany(std => std.RecursiveGetNestedData()).ToList();
+            return serializables;
         }
 
         #region Unused legacy code
 
         // ======== UNUSED =======
```

Comment:
There is a pretty big block of "unused" code here that we should delete. Not sure what we're keeping it around for.